### PR TITLE
Enhancement/user crud

### PR DESCRIPTION
**Pull request by Yunnigg**

### CODE REVIEW

File: pronto_mia/lib/core/queries/user_queries.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		      canEditDepartments: true
+        }
+      ) {
+        id
+      }
+    }
+  """;
+
+  static const updateUser = """
+     mutation updateUser(\$id: Int!, \$userName: String!, \$password: String!) {
```
Comment (Vec7or) [inline pronto_mia/lib/core/queries/user_queries.dart:43]: Here you could also update the department as well as the acl.

File: pronto_mia/lib/ui/views/user/edit/user_edit_viewmodel.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  NavigationService get _navigationService => locator.get<NavigationService>();
+
+  final User user;
+  final bool isDialog;
+
+  UserEditViewModel({this.user, this.isDialog = false});
+
+  @override
+  void setFormStatus() {
+    clearErrors();
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/user/edit/user_edit_viewmodel.dart:27]: 👍🏽

**Review by Vec7or — APPROVED** [review]

File: pronto_mia/lib/core/models/profiles.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+class Profile {
+  final String description;
+  final AccessControlList accessControlList;
+
+  const Profile({this.description, this.accessControlList});
+}
+
+class Profiles {
+  static const Profile administrator = Profile(
```
[the remark this answers is not in the document]
Comment (Yunnigg) [reply pronto_mia/lib/core/models/profiles.dart:11]: I think i will make a map out of it. That way it's iterable and the elements are accessible by their names.

File: pronto_mia/lib/core/queries/user_queries.dart (hunk 1 above)
Comment (Yunnigg) [reply pronto_mia/lib/core/queries/user_queries.dart:43]: Does that mean when I update the User with a new department it is not added to the users list of the department?

Comment (Vec7or) [reply pronto_mia/lib/core/queries/user_queries.dart:43]: It is added. But in your operation there is no department neither a acl present which could be set.

**Pull request merged**